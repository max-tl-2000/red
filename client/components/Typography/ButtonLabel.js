/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import createElement from './create-element';

const Text = createElement('button-label');

export default Text;
