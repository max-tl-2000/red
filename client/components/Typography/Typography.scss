@import '../../sass/_colors.scss';
@import '../../sass/_typography-mixins.scss';

.text-heavy {
  @include bodyHeavy();
}

.headline {
  @include headline();
}

.subheader {
  @include subHeader();
}

.title {
  @include textTitle();
}

.text {
  @include body();
}

.caption {
  @include caption();
}

.link {
  @include link();
}

.button-label {
  @include textButton();
}

.text-heavy,
.headline,
.subheader,
.title,
.text,
.caption {
  &.bold {
    font-weight: 500;
  }

  &.secondary {
    @include textSecondary();
  }

  &.highlight {
    @include lightHighlight();
  }

  &.error {
    @include lightError();
  }

  &.disabled {
    @include textDisabled();
  }

  &.lighter {
    @include textPrimary(light);

    &.secondary {
      @include textSecondary(light);
    }

    &.highlight {
      @include lightHighlight(light);
    }

    &.error {
      @include lightError(light);
    }

    &.disabled {
      @include textDisabled(light);
    }
  }

  &.uppercase {
    text-transform: uppercase;
  }

  &.errorUnderline {
    border-bottom: dotted 2px #ff1744;
  }

  &.noDefaultColor,
  &.lighter.noDefaultColor {
    color: inherit;
  }
}

.link {
  cursor: pointer;

  &.noDefaultColor {
    color: inherit;
  }

  &.underline {
    text-decoration: underline;
  }

  &.uppercase {
    text-transform: uppercase;
  }

  &.bold {
    font-weight: bold;
  }
}

.ellipsis {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.formattedBlock {
  .text-heavy,
  .headline,
  .subheader,
  .title,
  .text,
  .caption {
    margin-bottom: .8em;
  }
}
