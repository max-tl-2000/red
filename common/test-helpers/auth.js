/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import moment from 'moment'; // eslint-disable-line red/no-moment
import { expect } from 'chai';

export const verifyTokenExpiration = (decodedToken, expectedTokenExpiration) => {
  const iat = moment.unix(decodedToken.iat);
  const exp = moment.unix(decodedToken.exp);

  const result = exp.diff(iat, expectedTokenExpiration.substring(expectedTokenExpiration.length - 1));
  const expectedValue = expectedTokenExpiration.substring(0, expectedTokenExpiration.length - 1);
  expect(result).to.equal(+expectedValue);
};
