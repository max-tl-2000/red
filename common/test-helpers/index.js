/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
import { unmountComponentAtNode } from 'react-dom';
import when from './when';
import and from './and';
import * as overrider from './overrider';
import * as sandboxer from './sandboxer';
import sleep from '../helpers/sleep';

sandboxer.setSinon(sinon);

const expect = chai.expect;
const assert = chai.assert;

chai.use(sinonChai);

let div;

if (global.document) {
  beforeEach(() => {
    div = document.createElement('div');
    div.setAttribute('id', 'sandbox');

    document.body.appendChild(div);
  });
}

afterEach(() => {
  if (global.document && div) {
    unmountComponentAtNode(div);
    document.body.removeChild(div);
    div = null;
  }

  overrider.restore();
  sandboxer.restore();
});

const getSandBoxDiv = () => div;

// helper method to trigger a real click event in the browser
// jquery.fn.click is not the same as this as the event need
// to also generate the capture phase
const triggerMouseEvent = (ele, evt) => {
  const mEvent = document.createEvent('MouseEvent');
  // interface
  // event.initMouseEvent(type, canBubble, cancelable, view,
  //                    detail, screenX, screenY, clientX, clientY,
  //                    ctrlKey, altKey, shiftKey, metaKey,
  //                    button, relatedTarget);
  mEvent.initMouseEvent(evt, true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);

  ele.dispatchEvent(mEvent);
};

export { getSandBoxDiv, triggerMouseEvent, overrider, expect, assert, sinon, when, sandboxer, and, sleep };
