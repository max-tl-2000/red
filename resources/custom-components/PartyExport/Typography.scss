@import '../../../client/sass/colors';
@import '../../../client/sass/typography-mixins';

.text-heavy {
  @include bodyHeavy();
}

.headline {
  @include headline();
}

.subheader {
  @include subHeader();
}

.title {
  @include textTitle();
}

.text {
  @include body();
}

.button-label {
  @include textButton();
}

.caption {
  @include caption();
}

.link {
  @include link();
}

.text-heavy,
.headline,
.subheader,
.title,
.text,
.caption {
  &.bold {
    font-weight: 500;
  }

  &.secondary {
    @include textSecondary();
  }

  &.highlight {
    @include lightHighlight();
  }

  &.error {
    @include lightError();
  }

  &.disabled {
    @include textDisabled();
  }

  &.lighter {
    @include textPrimary(light);

    &.secondary {
      @include textSecondary(light);
    }

    &.highlight {
      @include lightHighlight(light);
    }

    &.error {
      @include lightError(light);
    }

    &.disabled {
      @include textDisabled(light);
    }
  }

  &.uppercase {
    text-transform: uppercase;
  }

  &.errorUnderline {
    border-bottom: dotted 2px #ff1744;
  }

  &.noDefaultColor,
  &.lighter.noDefaultColor {
    color: inherit;
  }
}

.ellipsis {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
