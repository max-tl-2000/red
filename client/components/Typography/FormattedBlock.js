/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import React from 'react';
import { cf, g } from './Typography.scss';

const FormmatedBlock = ({ id, className, children, ...rest }) => (
  <section id={id} className={cf('formattedBlock', g(className))} {...rest}>
    {children}
  </section>
);

export default FormmatedBlock;
