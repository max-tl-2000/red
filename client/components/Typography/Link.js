/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import React from 'react';
import { cf, g } from './Typography.scss';

const Link = ({ id, className, uppercase, noDefaultColor, underline, bold, href, children, ...rest }) => (
  <a href={href} id={id} className={cf('link', { noDefaultColor, underline, uppercase, bold }, g(className))} {...rest}>
    <span>{children}</span>
  </a>
);

export default Link;
