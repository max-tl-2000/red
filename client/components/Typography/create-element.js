/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import PropTypes from 'prop-types';
import { createElement } from 'react';
import { cf, g } from './Typography.scss';
import { isValidChildrenProp } from './validate-content';

/**
 * create Typography element, since all of them share the same interface and the only thing
 * that really change is the name of the element, this was abstracted in order to avoid
 * code duplication
 *
 * @param      {string}  nameOfElement  The name of element
 * @return     {ReactElement}  A Typography element
 */
const elementCreator = nameOfElement => {
  const Element = ({
    className,
    raw,
    noDefaultColor,
    secondary,
    bold,
    errorUnderline,
    ellipsis,
    highlight,
    error,
    disabled,
    inline,
    children,
    lighter,
    uppercase,
    testId,
    ...props
  }) => {
    const theProps = {
      'data-id': testId,
      'data-component': nameOfElement,
      'data-typography-element': true,
      className: cf(
        nameOfElement,
        raw
          ? {}
          : {
              noDefaultColor,
              secondary,
              bold,
              highlight,
              ellipsis,
              error,
              lighter,
              errorUnderline,
              disabled,
              uppercase,
            },
        g(className),
      ),
      ...props,
    };

    const tagElement = inline ? 'span' : 'p';
    // some elements report typeof 'object' but are in reality arrays
    if (!isValidChildrenProp(children)) {
      console.error('Invalid children found', nameOfElement, children);
      children = ''; // just handle it as an empty text
    }

    return createElement(tagElement, theProps, children);
  };

  Element.displayName = nameOfElement;

  Element.propTypes = {
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.bool, PropTypes.number, PropTypes.element, PropTypes.array]),
    className: PropTypes.string,
    noDefaultColor: PropTypes.bool,
    secondary: PropTypes.bool,
    bold: PropTypes.bool,
    errorUnderline: PropTypes.bool,
    ellipsis: PropTypes.bool,
    highlight: PropTypes.bool,
    error: PropTypes.bool,
    disabled: PropTypes.bool,
    inline: PropTypes.bool,
    lighter: PropTypes.bool,
    uppercase: PropTypes.bool,
    raw: PropTypes.bool,
  };

  return Element;
};

export default elementCreator;
