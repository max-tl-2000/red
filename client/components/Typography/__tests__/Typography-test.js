/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { mount } from 'enzyme';
import React from 'react';
import * as T from '../Typography';
import { restore, override } from '../../../../common/test-helpers/overrider';

describe('Typography', () => {
  let consoleMock;

  beforeEach(() => {
    consoleMock = { error: jest.fn() };

    override(console, consoleMock);
  });

  afterEach(() => restore());

  it('should not throw when an object is provided as children', () => {
    const testFn = () => {
      mount(<T.Text>{{ hello: 'world' }}</T.Text>);
    };
    expect(testFn).not.toThrow();
    expect(consoleMock.error).toHaveBeenCalledTimes(2);
    expect(consoleMock.error).toHaveBeenLastCalledWith('Invalid children found', 'text', {
      hello: 'world',
    });
  });

  it('should render simple Text elements', () => {
    const component = mount(<T.Text>Some simple text element</T.Text>);
    expect(component).toMatchSnapshot();
  });
});
