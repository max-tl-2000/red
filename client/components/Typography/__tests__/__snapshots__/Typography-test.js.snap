// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render simple Text elements 1`] = `
<text>
  <p
    className="text"
    data-component="text"
    data-typography-element={true}
  >
    Some simple text element
  </p>
</text>
`;
